lib/src/testing_library/react/react_element_queries.dart

```diff
@@ -0,0 +1,1178 @@
+import 'dart:html' show Element, ImageElement, InputElement, LabelElement, SelectElement, TextAreaElement;
+
+import 'package:over_react_test/src/testing_library/dom/async/types.dart' show MutationObserverInit;
+import 'package:over_react_test/src/testing_library/dom/matches/types.dart' show NormalizerFn, NormalizerOptions;
+import 'package:over_react_test/src/testing_library/dom/queries.dart' show DomQueries;
+import 'package:over_react_test/src/testing_library/react/render/render.dart' show currentRtlRenderContainer, render;
+
+// ----------------------------------------------------
+//  Internals (private)
+// ----------------------------------------------------
+
+/// Queries made available using the [currentRtlRenderContainer] made available after rendering
+/// something using [render].
+///
+/// None of the methods in here are accessed directly by consumers, and are instead exposed
+/// as top-level functions to mimic the API provided by the react-testing-library.
+///
+/// > Related: `screen`, `within()`
+// PRIVATE. DO NOT EXPORT.
+class _ReactElementQueries extends DomQueries {
+  _ReactElementQueries() : super(() => currentRtlRenderContainer);
+}
+
+final _queries = _ReactElementQueries();
```

Comment:
This is how I went about ensuring that consumers wouldn't have to specify the default container that the `render()` function renders into when calling the "top level" queries that follow below.

Reply:
Unfortunately, this also required a 1:1 mapping between the methods defined in the query mixins, and a top-level function which just proxies a call to the mixin method. So, a little extra boilerplate - but a nice API for the consumer IMO.